doc/api/util.md

```diff
@@ -1097,6 +1097,80 @@ console.log(values, positionals);
 // Prints: [Object: null prototype] { foo: true, bar: 'b' } []
 ```
 
+Detailed parse information is available for adding custom behaviours by
+specifying `tokens: true` in the configuration. The returned tokens have
+properties describing:
+
+* all tokens
+  * `kind` { string } One of 'option', 'positional', or 'option-terminator'.
+  * `index` { number } Index of element in `args` containing token.
+* option tokens
+  * `name` { string } Long name of option.
+  * `rawName` { string } How option used in args, like `-f` of `--foo`.
+  * `value` { string | undefined } Option value specified in args.
+    Undefined for boolean options.
+  * `inlineValue` { boolean | undefined } Whether option value specified inline,
+    like `--foo=bar`.
+* positional tokens
+  * `value` { string } Positional value (i.e. `args[index]`).
+* option-terminator token
+
+For example, assuming the following script which uses
+automatic detection of options and no error checking:
+
+```mjs
+import { parseArgs } from 'node:util';
+console.log(parseArgs({ strict: false, tokens: true }));
+```
+
+```cjs
+const { parseArgs } = require('node:util');
+console.log(parseArgs({ strict: false, tokens: true }));
+```
+
+This call shows the three kinds of token and their properties:
+
+```console
+$ node tokens.cjs -xy --foo=BAR -- file.txt
+{
+  values: [Object: null prototype] { d: true, foo: 'BAR' },
+  positionals: [ 'file.txt' ],
+  tokens: [
+    {
+      kind: 'option',
+      name: 'x',
+      rawName: '-x',
```

Comment:
Since we have `name` to point towards the actual option that this token corresponds to, wouldn't it potentially provide more flexibility if `rawName` was the value `-xy`.

I think this was discussed elsewhere, but I didn't fully understand the reasoning for not storing the full parsed value.

Reply:
What you want is the ability to determine the form of the original argument, so that you can do stuff like `` console.error(`The argument to ${token.rawName} was malformed`) ``. Exposing `-x` makes that trivial; exposing `-xy` makes that difficult.

`-xy` is pretty much never useful, as far as I can tell. How do you imagine it being used?

Discussion is [here](https://github.com/pkgjs/parseargs/pull/129#discussion_r884576611).

Reply:
I could imagine it being useful if you wanted to parse something like:

-vvv, for different levels of log verbosity, or -n10 to assign the value 10 to an option.

Reply:
For `-vvv`, you'll get three `{ kind: 'option', name: 'verbose' }` tokens, so you don't need to parse them yourself - you can just count the tokens whose name is `verbose` (though the recommended way to do that would be with `multiple`, and then check the length of the resulting array in `values`).

And for `-n10` you'll get a `{ kind: 'option', name: 'n', value: '10' }` token. (Plus of course the `values` object will have `{ n: '10' }`; you shouldn't normally need to parse it out of the tokens.)

Reply:
> I think this was discussed elsewhere, but I didn't fully understand the reasoning for not storing the full parsed value.

It could be stored as well, but not instead.

Why didn't we add unparsed argument like `-xy`?  The intention is that most processing can be done with the included meta-data without doing ad-hoc parsing of the original argument. The unparsed argument is available fairly easily using `args[token.index]`.

Why not offer `-xy` _instead_ of `-x`? The `rawName` property describes how the command-line user used the option. It is useful for error messages, and surprisingly hard to work out robustly (strict vs loose, options with inline values, short option groups).